main.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+def bubble_sort(arr)
+  (0..arr.length - 1).each do |i|
+    (1..(arr.length - 1) - i).each do |j|
+      arr[j], arr[j - 1] = arr[j - 1], arr[j] if arr[j - 1] > arr[j]
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  (0..arr.length - 1).each do |i|
+    (1..(arr.length - 1) - i).each do |j|
+      if block_given?
+        arr[j], arr[j - 1] = arr[j - 1], arr[j] if yield(arr[j - 1], arr[j]).positive?
+      elsif arr[j - 1] > arr[j]
+        arr[j], arr[j - 1] = arr[j - 1], arr[j]
```

Comment:
- [ ] If a block is not given, this function should raise an error

Reply:
@samgaco I've revised the code to include an error message for missing block. I'll submit a separate code review

Reply:
@samgaco 
Thank you for taking the time to review our code

😄